sky/serve/serve_utils.py

```diff
@@ -356,20 +357,32 @@ def wait_service_initialization(service_name: str, job_id: int) -> str:
     cnt = 0
     while True:
         record = serve_state.get_service_from_name(service_name)
-        if record is None:
-            continue
-        if job_id != record['controller_job_id']:
-            return common_utils.encode_payload(None)
-        lb_port = record['load_balancer_port']
-        if lb_port is not None:
-            return common_utils.encode_payload(lb_port)
+        if record is not None:
+            if job_id != record['controller_job_id']:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'The service {service_name!r} is already running. '
+                        'Please specify a different name for your service. '
+                        'To update an existing service, run: `sky serve down` '
+                        'and then `sky serve up` again (in-place update will '
+                        'be supported in the future).')
+            lb_port = record['load_balancer_port']
+            if lb_port is not None:
+                return common_utils.encode_payload(lb_port)
+        elif len(serve_state.get_services()) >= NUM_SERVICE_THRESHOLD:
```

Comment:
Should we only check the non-finished services instead?

Reply:
Wdym by non-finished? The current implementation will still cleanup the service once the user `sky serve down` it.

Reply:
Oh, I mean the services in FAILED states -- I feel like asking the user to manually `sky serve down` a FAILED service feels a bit weird, but I guess it should be fine for now. We can wait for user's feedback for this. : )

Reply:
For services in FAILED states, we still keep it running so it still takes some memory. The reason is that we will support update services, which could convert them to be READY 🤔